ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -1984,14 +1977,270 @@ const triggerSet: TriggerSet<Data> = {
+      netRegex: { npcNameId: superchainNpcNameId, npcBaseId: superchainNpcBaseIds, capture: false },
+      condition: (data) => data.phase === 'superchain2b' && data.superchainCollect.length === 4,
+      alertText: (data, _matches, output) => {
+        // Sort ascending. collect: [dest1, dest2, out/sphere, in/donut]
+        const collect = data.superchainCollect.slice(0, 4).sort((a, b) =>
+          parseInt(a.npcBaseId) - parseInt(b.npcBaseId)
+        );
+
+        const donut = collect[3];
+        if (donut === undefined)
+          return;
+
+        // For the first mechanic, two destination orbs span at [100,95] and [100,105]
+        // Each has a short tether to either an 'in' or 'out' orb on the same N/S half of the area.
+        // We therefore only need to know whether the 'in' orb is N or S to identify the safe spot.
+        return parseFloat(donut.y) > 100 ? output.south!() : output.north!();
+      },
+      outputStrings: {
+        north: Outputs.north,
+        south: Outputs.south,
+      },
+    },
+    {
+      id: 'P12S Superchain Theory IIb Second Mechanic',
+      type: 'AddedCombatant',
+      netRegex: { npcNameId: superchainNpcNameId, npcBaseId: superchainNpcBaseIds, capture: false },
+      condition: (data) => data.phase === 'superchain2b' && data.superchainCollect.length === 8,
+      delaySeconds: 6.5,
+      durationSeconds: 6, // keep active until just before Ray of Light 2
+      alertText: (data, _matches, output) => {
+        // Sort ascending. collect: [dest1, dest2, out, partnerProtean]
+        const collect = data.superchainCollect.slice(4, 8).sort((a, b) =>
+          parseInt(a.npcBaseId) - parseInt(b.npcBaseId)
+        );
+
+        const partnerProtean = collect[3];
+        if (partnerProtean === undefined)
+          return;
+        let mechanicStr;
+        if (partnerProtean.npcBaseId === superchainNpcBaseIdMap.protean) {
+          mechanicStr = output.protean!();
+          data.superchain2bSecondMech = 'protean';
+        } else {
+          mechanicStr = output.partners!();
+          data.superchain2bSecondMech = 'partners';
+        }
+
+        // For the second mechanic, the two destination orbs spawn at [92,100] and [108,100]
+        // One is tethered to a sphere (out) orb, and the other to a partner or protean orb.
+        // The partner/protean orb is always on the same E/W half as the destination orb it is tethered to.
+        // We therefore only need to know whether the partnerProteam orb is E or W to identify the safe spot.
+        const x = parseFloat(partnerProtean.x);
+        if (x > 100) {
+          data.superchain2bSecondDir = 'east';
+          return output.combined!({ dir: output.east!(), mechanic: mechanicStr });
+        }
+        data.superchain2bSecondDir = 'west';
+        return output.combined!({ dir: output.west!(), mechanic: mechanicStr });
+      },
+      outputStrings: {
+        combined: {
+          en: '${dir} (Side) => ${mechanic} After',
```

Comment:
Can this say left/west => mechanic, since most folks are standing north/south, looking at the boss and a left/right call could helpful.

I also personally think having this call earlier would be better so folks know which direction they can already start moving since the donut is big.